hdfs-dataset-sink/src/test/java/org/springframework/cloud/stream/module/dataset/sink/DatasetSinkIntegrationTests.java

```diff
@@ -0,0 +1,221 @@
+import org.springframework.data.hadoop.store.dataset.DatasetOperations;
+import org.springframework.messaging.support.GenericMessage;
+import org.springframework.messaging.support.MessageBuilder;
+import org.springframework.test.annotation.DirtiesContext;
+import org.springframework.test.context.junit4.SpringJUnit4ClassRunner;
+
+import java.io.File;
+import java.io.FilenameFilter;
+import java.io.IOException;
+import java.text.SimpleDateFormat;
+
+/**
+ * @author Thomas Risberg
+ */
+@RunWith(SpringJUnit4ClassRunner.class)
+@SpringApplicationConfiguration(classes = DatasetSinkApplication.class)
+@DirtiesContext
+public abstract class DatasetSinkIntegrationTests {
+
+	@Autowired
+	ConfigurableApplicationContext applicationContext;
+
+	@Value("${directory}#{T(java.io.File).separator}${namespace}")
+	protected String testDir;
+
+	@Autowired
+	protected FsShell fsShell;
+
+	@Autowired
+	protected DatasetOperations datasetOperations;
+
+	@Autowired
+	@Bindings(DatasetSink.class)
+	protected Sink sink;
+
+	@Before
+	public void setup() {
+		if (fsShell.test(testDir)) {
+			fsShell.rmr(testDir);
+		}
+	}
+
+	@WebIntegrationTest({"server.port:0",
+			"spring.hadoop.fsUri=file:///",
+			"directory=${java.io.tmpdir}/dataset",
+			"namespace=test",
+			"batchSize=2",
+			"idleTimeout=2000"})
+	public static class WritingDatasetTests extends DatasetSinkIntegrationTests {
+
+		@Test
+		public void testWritingSomething() throws IOException {
+			sink.input().send(new GenericMessage<>("Foo"));
+			sink.input().send(new GenericMessage<>("Bar"));
+			sink.input().send(new GenericMessage<>("Baz"));
+		}
+
+		@After
+		public void checkFilesClosedOK() throws IOException {
+			applicationContext.close();
+			File testOutput = new File(testDir);
+			assertTrue("Dataset path created", testOutput.exists());
+			assertTrue("Dataset storage created",
+					new File(testDir + File.separator + datasetOperations.getDatasetName(String.class)).exists());
+			assertTrue("Dataset metadata created",
+					new File(testDir + File.separator + datasetOperations.getDatasetName(String.class) +
+							File.separator + ".metadata").exists());
+			File testDatasetFiles =
+					new File(testDir + File.separator + datasetOperations.getDatasetName(String.class));
+			File[] files = testDatasetFiles.listFiles(new FilenameFilter() {
+				@Override
+				public boolean accept(File dir, String name) {
+					if (name.endsWith(".avro")) {
+						return true;
+					}
+					return false;
+				}
+			});
+			assertTrue("Dataset data files created", files.length > 1);
+		}
+	}
+
+	@WebIntegrationTest({"server.port:0",
+			"spring.hadoop.fsUri=file:///",
+			"directory=${java.io.tmpdir}/dataset",
+			"namespace=pojo",
+			"allowNullValues=false",
+			"format=avro",
+			"partitionPath=year('timestamp')",
+			"batchSize=2",
+			"idleTimeout=2000"})
+	public static class PartitionedDatasetTests extends DatasetSinkIntegrationTests {
+
+		Long systemTime = 1446570266690L; // Date is 2015/11/03
```

Comment:
What is the purpose of this?
Why `System.currentTimeMillis()` isn't enough if that?